src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/InstructionList.cs

```diff
@@ -209,6 +209,21 @@ private void UpdateStackDepth(Instruction instruction)
             }
         }
 
+        // "Un-emit" the previous instruction.
+        // Useful if the instruction was emitted in the calling method, and covers the more usual case.
+        // In particular, calling this after an EmitPush() or EmitDup() costs about the same as adding
+        // an EmitPop() to undo it at compile time, and leaves a slightly leaner instruction list.
+        public void Belay()
```

Comment:
Or simply `UnEmit` perhaps. (More generally, the antonym of *emit* is *absorb*, but that twists the metaphor into weird places).